src/parser/WASMParser.cpp

```diff
@@ -896,27 +896,24 @@ class WASMBinaryReader : public wabt::WASMBinaryReaderDelegate {
 
     virtual void OnGlobalGetExpr(Index index) override
     {
-        auto sz = Walrus::valueSizeInStack(m_result.m_globalTypes[index]->type());
-        auto stackPos = pushVMStack(sz);
-        if (sz == 4) {
+        auto type = m_result.m_globalTypes[index]->type();
+        auto stackSize = Walrus::valueSizeInStack(type);
+        auto stackPos = pushVMStack(stackSize);
+        if (type == Walrus::Value::Type::I32 || type == Walrus::Value::Type::F32) {
             pushByteCode(Walrus::GlobalGet32(stackPos, index));
         } else {
-            ASSERT(sz == 8);
             pushByteCode(Walrus::GlobalGet64(stackPos, index));
         }
     }
 
     virtual void OnGlobalSetExpr(Index index) override
     {
+        auto type = m_result.m_globalTypes[index]->type();
+        auto stackSize = Walrus::valueSizeInStack(type);
         auto stackPos = peekVMStack();
-
-        auto sz = Walrus::valueSizeInStack(m_result.m_globalTypes[index]->type());
-        if (sz == 4) {
-            ASSERT(peekVMStackSize() == 4);
+        if (type == Walrus::Value::Type::I32 || type == Walrus::Value::Type::F32) {
             pushByteCode(Walrus::GlobalSet32(stackPos, index));
         } else {
-            ASSERT(sz == 8);
-            ASSERT(peekVMStackSize() == 8);
```

Comment:
Why did you remove these assertions?
Assertion is used to check the correct execution of program and it does not affect the overall performance in release mode